### PR TITLE
Borders - part 1

### PR DESCRIPTION
**Pull request by istarkov**
## Description

ref #1169

image

Based on this comment https://github.com/webstudio-is/webstudio-builder/issues/1169#issuecomment-1478260843
Border Style section is added. As no  section `+` button exists.

IconButton has changed to be in sync with design.
ToggleButten added.

To review start with borders.tsx

## Steps for reproduction

Open Borders section, change style, borders width, radius, individual radius, individual borders

## Code Review
- [x] hi @rpominov , I need you to do
check that ToggleButton component and its stories are ok.

- [x] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [x] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio-builder/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/builder/features/style-panel/sections/borders/border-property.tsx [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    batch.publish(options);
+  };
+
+  return (
+    <Grid gap={1}>
+      <Grid
+        css={{
+          gridTemplateColumns: individualModeIcon
+            ? "1fr 80px max-content"
+            : "1fr 116px",
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/sections/borders/border-property.tsx:120]: what are those hardcoded values? I don't think we should have any, either design is wrong or you used wrong layout or something, it should almost always fit some spacing token

File: apps/builder/app/builder/features/style-panel/sections/borders/border-style.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import type { StyleProperty } from "@webstudio-is/css-data";
+import type { RenderCategoryProps } from "../../style-sections";
+import { BorderProperty } from "./border-property";
+
+const borderPropertyOptions = {
+  borderTopStyle: {},
+  borderRightStyle: {},
+  borderLeftStyle: {},
+  borderBottomStyle: {},
+} as const satisfies Partial<{ [property in StyleProperty]: unknown }>;
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/sections/borders/border-style.tsx:10]: need to remember this thing

Comment (kof) [reply apps/builder/app/builder/features/style-panel/sections/borders/border-style.tsx:10]: isn't value known though?

File: apps/builder/app/builder/features/style-panel/sections/borders/borders.tsx [hunk 3]
```diff
@@ -1,19 +1,59 @@
...
 }: RenderCategoryProps) => {
   return (
-    <>
-      {styleConfigsByCategory.map((entry) => renderProperty(entry))}
-      <ShowMore
-        styleConfigs={moreStyleConfigsByCategory.map((entry) =>
-          renderProperty(entry)
-        )}
+    <Flex direction="column" gap={2}>
+      <Grid css={{ gridTemplateColumns: "1fr 116px" }}>
```
Comment (kof) [inline apps/builder/app/builder/features/style-panel/sections/borders/borders.tsx:20]: discussed in dm, we we can get rid of that hardcoded value

**Review by kof — APPROVED** [review]

File: apps/builder/app/builder/features/style-panel/sections/borders/border-style.tsx (hunk 2 above)
Comment (istarkov) [reply apps/builder/app/builder/features/style-panel/sections/borders/border-style.tsx:10]: Here I just check that property names are from StyleProperty values.
The value of property will be checked by react component. 
But no way to check on Component side that no additional properties will be added.

Example:

https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAKjgQwM5wEoFNkGN4BmUEIcA5FDvmQNwBQduEAdqvAMImTNbPwC8cABRhiYVAC44AbwhgYwFpMxYmUACYAeMsjJwAPuQBGZADRxmAVxBGsUAHxwAvgEo4-R9LpwfcSjEsoZgtLABtQ+idGJXg5BSV3GWQJAEZzIyk0uFxM8ywpACYnFHQmVhh6OgB6KrgAOQg4O2IoOAALOyw6TU5wFl5Y+UVWRNkhpWKq+wYauAAJTrgASTgAcyx4ZGDm6GjyuDjh1ALR5KyMuCycy7zC4rRsmLhUZAVUAmAsdGw1LR09QxkEzmSzMADWzAgAHdmPZ6EA

So with satisfies I want to show that I don't mind what values this component has, moreover don't want to create types for values, but keys must be real StyleProperty.

Also satisfies forces intellisense.

File: packages/design-system/src/components/toggle-button.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import { forwardRef, type ComponentProps } from "react";
+import type { CSS } from "../stitches.config";
+import * as TogglePrimitive from "@radix-ui/react-toggle";
+import { IconButton } from "./icon-button";
+
+type Props = {
+  variant?: Extract<
+    ComponentProps<typeof IconButton>["variant"],
+    { "@initial"?: unknown }
+  >["@initial"];
```
Comment (rpominov) [inline packages/design-system/src/components/toggle-button.tsx:15]: What does this code do? :)

I wouldn't use this weird `@initial` thing from stitches, would rather just define type as `"something" | "something1"` here

File: packages/design-system/src/components/icon-button.tsx
Comment (rpominov) [inline packages/design-system/src/components/icon-button.tsx:1]: Please add a header 

/**
 * Implementation of the "Icon Button" component from:
 * https://www.figma.com/file/sfCE7iLS0k25qCxiifQNLE/%F0%9F%93%9A-Webstudio-Library?node-id=4-3139
 *
 * Also used as "Toggle Button"
 */

**Comment by TrySound** [comment]
Please remove fill=none from icons. They are invisible as .svg

File: apps/builder/app/dashboard/projects/project-card.tsx [hunk 5]
```diff
@@ -100,7 +100,6 @@ const Menu = ({
     <DropdownMenu open={isOpen} onOpenChange={setIsOpen}>
       <DropdownMenuTrigger asChild>
         <IconButton
-          variant={isOpen ? "active" : "default"}
```
[the remark this answers is not in the document]
Comment (istarkov) [reply apps/builder/app/dashboard/projects/project-card.tsx:103]: see data-* rules, they active coming from radix

**Pull request merged**